jmxtrans-output/jmxtrans-output-core/src/test/java/com/googlecode/jmxtrans/model/output/OpenTSDBWriterTests.java

```diff
@@ -92,201 +96,89 @@ public void	setupTest () throws Exception {
 
-		try {
-			// Execute.
-			this.writer.start();
-			Assert.fail("LifecycleException missing");
-		} catch ( LifecycleException lcExc ) {
-			// Verify.
-			Assert.assertSame(ioExc, lcExc.getCause());
-			Mockito.verify(this.mockLog).error(contains("opening socket"), eq(ioExc));
-		}
-	}
+		Mockito.when(socket.getOutputStream()).thenReturn(out);
 
-	/**
-	 * Test throwing of IOException when closing the socket.
-	 */
-	@Test
-	public void	testSocketCloseIOException () throws Exception {
-		// Prepare.
-		IOException	ioExc = new IOException("X-TEST-IO-EXC-X");
-		Mockito.doThrow(ioExc).when(this.mockSocket).close();
+		OpenTSDBWriter writer = OpenTSDBWriter.builder()
+				.setHost("localhost")
+				.setPort(4243)
+				.addTypeNames(typeNames)
+				.build();
+		writer.setPool(pool);
 
-		// Execute.
-		this.writer.start();
-		try {
-			this.writer.stop();
-			Assert.fail("LifecycleException missing");
-		} catch ( LifecycleException lcExc ) {
-			// Verify.
-			Assert.assertSame(ioExc, lcExc.getCause());
-			Mockito.verify(this.mockLog).error(contains("closing socket"), eq(ioExc));
-		}
+		return writer;
 	}
 
-	/**
-	 * Test throwing of IOException when starting output.
-	 */
 	@Test
-	public void	testStartOutputIOException () throws Exception {
-		// Prepare.
-		IOException	ioExc = new IOException("X-TEST-IO-EXC-X");
-		Mockito.when(this.mockSocket.getOutputStream()).thenThrow(ioExc);
+	public void writeSingleResult() throws Exception {
+		Server server = Server.builder().build();
+		Query query = Query.builder().build();
+		Result result = new Result(System.currentTimeMillis(), "attributeName", "className", "objDomain", "classNameAlias", "typeName", ImmutableMap.of("key", (Object)1));
 
-		// Execute.
-		this.writer.start();
-		try {
-			this.writer.doWrite(null, this.mockQuery, ImmutableList.of(this.mockResult));
-			Assert.fail("IOException missing");
-		} catch ( IOException ioCaught ) {
-			// Verify.
-			Assert.assertSame(ioExc, ioCaught);
-			Mockito.verify(this.mockLog).error(contains("output stream"), eq(ioExc));
-		}
-	}
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		OpenTSDBWriter writer = getOpenTSDBWriter(out);
 
-	/**
-	 * Test IOException thrown on SendOutput.
-	 */
-	@Test
-	public void	testSendOutputIOException () throws Exception {
-		// Prepare.
-		IOException	ioExc = new IOException("X-TEST-IO-EXC-X");
-		Mockito.doThrow(ioExc).when(this.mockOut).writeBytes(anyString());
+		writer.doWrite(server, query, of(result));
 
-		// Execute.
-		this.writer.start();
-		try {
-			this.writer.doWrite(null, this.mockQuery, ImmutableList.of(this.mockResult));
-			Assert.fail("IOException missing");
-		} catch ( IOException ioCaught ) {
-			// Verify.
-			Assert.assertSame(ioExc, ioCaught);
-			Mockito.verify(this.mockLog).error(contains("writing result"), eq(ioExc));
-		}
+		// check that OpenTSDB format is respected
+		Assertions.assertThat(out.toString()).startsWith("put classNameAlias.attributeName ")
```

Comment:
Static import of `Assertions.assertThat()` would be more readable (if there is no conflict with another assertion framework).